server/src/test/java/org/elasticsearch/search/aggregations/CardinalityUpperBoundTests.java

```diff
@@ -22,20 +22,46 @@
 import org.elasticsearch.test.ESTestCase;
 
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.sameInstance;
 
 public class CardinalityUpperBoundTests extends ESTestCase {
     public void testNoneMultiply() {
-        assertThat(CardinalityUpperBound.NONE.multiply(randomInt()), equalTo(CardinalityUpperBound.NONE));
+        assertThat(CardinalityUpperBound.NONE.multiply(randomInt()), sameInstance(CardinalityUpperBound.NONE));
+    }
+
+    public void testNoneMap() {
+        assertThat(CardinalityUpperBound.NONE.map(i -> i), equalTo(0));
     }
 
     public void testOneMultiply() {
-        assertThat(CardinalityUpperBound.ONE.multiply(0), equalTo(CardinalityUpperBound.NONE));
-        assertThat(CardinalityUpperBound.ONE.multiply(1), equalTo(CardinalityUpperBound.ONE));
-        assertThat(CardinalityUpperBound.ONE.multiply(between(2, Integer.MAX_VALUE)), equalTo(CardinalityUpperBound.MANY));
+        assertThat(CardinalityUpperBound.ONE.multiply(0), sameInstance(CardinalityUpperBound.NONE));
+        assertThat(CardinalityUpperBound.ONE.multiply(1), sameInstance(CardinalityUpperBound.ONE));
+        assertThat(CardinalityUpperBound.ONE.multiply(Integer.MAX_VALUE), sameInstance(CardinalityUpperBound.MANY));
+    }
+
+    public void testOneMap() {
+        assertThat(CardinalityUpperBound.ONE.map(i -> i), equalTo(1));
+    }
+
+    public void testLargerKnownValues() {
+        int estimate = between(2, Short.MAX_VALUE);
+        CardinalityUpperBound known = CardinalityUpperBound.ONE.multiply(estimate);
+        assertThat(known.map(i -> i), equalTo(estimate));
+
+        assertThat(known.multiply(0), sameInstance(CardinalityUpperBound.NONE));
+        assertThat(known.multiply(1), sameInstance(known));
+        assertThat(known.multiply(between(Short.MAX_VALUE, Integer.MAX_VALUE)), sameInstance(CardinalityUpperBound.MANY));
```

Comment:
this looks  wrong to me.  In the minimum, `known` can have an estimate of 2, and 2 * `Short.MAX_VALUE` is only 2^16, isn't it?  I know CI passed it, possible it just got lucky on picking numbers.  But if this passes most of the time, I think something is wrong.

Reply:
Yeah, we got lucky. Looks like you have to iterate a lot to get unlucky, but my math was wrong.